portfolio/src/main/webapp/script.js

```diff
@@ -46,25 +46,19 @@ function grow() {
   };
   const image = document.getElementById('basil');
   console.log(`Let's grow some basil`);
-
-  function updateImagePerSec(imgIndex) {
-    setTimeout(function () {
-      console.log('Some time has passed ...');
-      // Change the image
-      const imgSrc = `images/basil${imgIndex}.jpg`;
-      image.src = imgSrc;
-      image.alt = srcToAlt[imgSrc];
-      image.width = '400';
-      image.height = '400';
-      imgIndex++;
-      if (imgIndex <= 6) {
-        updateImagePerSec(imgIndex); // Keep growing
-      } else {
-        console.log(`This's the latest photo I have! The basil will keep growing ...`);
-      }
-    }, 1000); // Update the image every 1000 ms = 1 sec
+  
+  for (let imgIndex = 1; imgIndex <= 6; imgIndex++) {
+    updateImg(imgIndex);
```

Comment:
This loop body looks like a synchronous call to me. In other words,
it looks like the loop updates the images one after another. Really
really it sets timeouts one after another, but we need to dig into
the implementation of updateImg to see that.

I think it would be clearer to pull the setTimeout call out of the
updateImg function. Something like:

Reply:
I agree that does make the asynchronous call more obvious to the reader. Thanks for the advice! I added a patch.